ark-activemq/templates/deployment.yaml

```diff
@@ -0,0 +1,124 @@
+            - name: ACTIVEMQ_SUNJMX_START
+              value: "-javaagent:/app/jmx_prometheus_javaagent.jar={{ .Values.ports.metrics }}:/app/jmx-prometheus-config.yaml"
+            {{- end }}
+          ports:
+            {{- if .Values.ports.console }}
+            - name: console
+              protocol: TCP
+              containerPort: {{ .Values.ports.console }}
+            {{- end }}
+            {{- if .Values.ports.amqp }}
+            - name: amqp
+              protocol: TCP
+              containerPort: {{ .Values.ports.amqp }}
+            {{- end }}
+            {{- if .Values.ports.mqtt }}
+            - name: mqtt
+              protocol: TCP
+              containerPort: {{ .Values.ports.mqtt }}
+            {{- end }}
+            {{- if .Values.ports.openwire }}
+            - name: openwire
+              protocol: TCP
+              containerPort: {{ .Values.ports.openwire }}
+            {{- end }}
+            {{- if .Values.ports.websocket }}
+            - name: websocket
+              protocol: TCP
+              containerPort: {{ .Values.ports.websocket }}
+            {{- end }}
+            {{- if .Values.ports.stomp }}
+            - name: stomp
+              protocol: TCP
+              containerPort: {{ .Values.ports.stomp }}
+            {{- end }}
+          {{- $first_open_port := or .Values.ports.amqp .Values.ports.mqtt .Values.ports.openwire .Values.ports.websocket .Values.ports.stomp }}
+          {{- if not $first_open_port }}
+            {{- fail "At least one of the AMQP, MQTT, Openwire, Websocket or STOMP port must be specified" }}
+          {{- end }}
+          readinessProbe:
+            tcpSocket:
+              port: {{ $first_open_port }}
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+          livenessProbe:
+            tcpSocket:
+              port: {{ $first_open_port }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            - name: data
+              mountPath: {{ .Values.persistence.mountPath }}
+              {{- if .Values.persistence.subPath }}
+              subPath: {{ .Values.persistence.subPath }}
+              {{- end }}
+            - name: secrets
+              mountPath: /app/conf
+              readOnly: true
+      volumes:
+        - name: data
+          persistentVolumeClaim:
+            claimName: {{ .Values.persistence.existingClaim }}
```

Comment:
@fabricetriboix  we should have a persistent volume claim created with helm. If we are using EBS we can go with dynamic provisioning..

Reply:
@naveen-armedia I think it's better to leave the creation of the persistent volume claim outside the Helm chart. A persistence volume claim necessarily makes assumptions on what is the environment it is running on, and this would prevent the Helm chart from being used in any environment.

For example, if I specify a persistence volume claim backed by EBS, we will only be able to deploy this Helm chart on AWS, and won't be able to deploy it in an on-premise environment or on developers' laptops. That is why I left the persistent volume claim outside the Helm chart, and I believe this is a best practice from the various Helm charts I have seen around.

Reply:
The persistent volume claim must be created by whatever is running the `helm create` command. Typically, that would be a script, such as [this script](https://github.com/ArkCase/DevOps/blob/develop/kubernetes/deploy.py) I started working on some time ago.

Reply:
@naveen-armedia We discussed this during our meeting on Friday. The persistent volume claim will be created and managed by the top-level `deploy.py` script. In that meeting, we decided that I would add ActiveMQ to that script, move the script to its own repository, and organize a meeting to show how it works.